tests/test_sasquatchDatastore.py

```diff
@@ -71,6 +72,24 @@ def test_put(self):
         mock_method.assert_called()
         self.assertIs(mock_method.call_args[0][0], bundle)
 
+    def test_explicit_timestamp_version(self):
+        dispatcher = SasquatchDispatcher("http://test.local", "na")
+        bundle = MetricMeasurementBundle()
+        bundle.timestamp_version = "explicit_timestamp"
+        # verify this raises with no specified time
+        with self.assertRaises(ValueError):
+            dispatcher._handleTimes({}, bundle, "localRun")
+        # verify this raise with a date that can't be parsed
+        bundle.timestamp_version = "explicit_timestamp:123233"
+        with self.assertRaises(ValueError):
+            dispatcher._handleTimes({}, bundle, "localRun")
+        # verify that a correct time gets parsed
+        bundle.timestamp_version = "explicit_timestamp:20230728T165102Z"
```

Comment:
I was confused as to why this works, as the [docs](https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes) seem to suggest that this should be `+0000`. However, I've tested it, and `strptime` in a notebook _does_ parse `Z` to `tzinfo=datetime.timezone.utc`. It doesn't parse any other time zone code (I.e., any other letter of the alphabet). Is `Z` just a special case?

Reply:
It is an explicit carve out: https://docs.python.org/3.11/library/datetime.html#technical-detail . This has been true formally since python 3.7 and reading various prs and such I think it implicitly worked prior to that and they just formalized it. Because it is the formal regex of the python datetime library, it will not change, as that would break so much backwards compatibility.